File: Hackerrank/DataStructures/linked_lists/insert_at_head.py
"""
 Insert Node at the begining of a linked list
 head input could be None as well for empty list
 Node is defined as
 
 class Node(object):
 
   def __init__(self, data=None, next_node=None):
       self.data = data
       self.next = next_node

 return back the head of the linked list in the below method. 
"""

def Insert(head, data):
    
    node = Node()
    node.data = data
    
    if head==None:
        head=node
        return head
    
    node.next=head
    head=node
    return head

File: Hackerrank/DataStructures/linked_lists/insert_at_head.cpp
/*
  Insert Node at the begining of a linked list
  Initially head pointer argument could be NULL for empty list
  Node is defined as 
  struct Node
  {
     int data;
     struct Node *next;
  }
return back the pointer to the head of the linked list in the below method.
*/
Node* Insert(Node *head,int data)
{
  // Complete this method
    Node* node = new Node();
    node->data = data;
    node->next = NULL;
    
    if (head==NULL){
        head = node;
        return(head);
    }
    
    node->next = head;
    head = node;
    return(head);
}